src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -215,7 +398,7 @@ def get_rid(self):
                 subscription=get_subscription_id(self.cmd.cli_ctx),
                 resource_group=self.resource_group.name,
                 namespace=CONTAINER_APPS_RP,
-                type="managedEnvironments",
+                type=self.resource_type if self.resource_type else MANAGED_ENVIRONMENT_TYPE,
```

Comment:
in initialization, you already set self.name to name of environment. In this way, with the context, we do not need to validate if it is valid resource id and we should always treat it as name, right?

Reply:
for command:
-name {appname} -environment {name}

if the container app and environment cannot be found,  resource_type is none, default to create MANAGED_ENVIRONMENT_TYPE resource only.